code/game/objects/items/wall_mounted.dm

```diff
@@ -58,24 +58,25 @@
 /obj/item/wallframe/proc/after_attach(obj/O)
 	transfer_fingerprints_to(O)
 
-/obj/item/wallframe/attackby(obj/item/W, mob/user, params)
-	..()
-	if(W.tool_behaviour == TOOL_SCREWDRIVER)
-		// For camera-building borgs
-		var/turf/T = get_step(get_turf(user), user.dir)
-		if(iswallturf(T))
-			T.attackby(src, user, params)
+/obj/item/wallframe/screwdriver_act(mob/living/user, obj/item/tool)
+	// For camera-building borgs
+	var/turf/T = get_step(get_turf(user), user.dir)
+	if(iswallturf(T))
+		T.attackby(user, tool)
```

Comment:
walls' attackby has a few helper procs for handling different behavior, this is probably going to stay like this.

Reply:
I'm not really a fan of this either, but it'd mean working down and fixing reinforced walls, which have a switch case for deconstruction and I feel like I'm getting deep enough into the weeds with this PR